### DISCUSSION TITLE
[expo-store-review] Determine if user actually clicked submit?

### DISCUSSION THREAD

**Discussion by zegenerative** [expo/expo#31342] [Proposals and Ideas]
Following the guidelines, a user should be prompted to review at certain [quota](https://developer.android.com/guide/playcore/in-app-review#quotas). 
They even suggest to extend this following your own approach (let's say one review per 6 months).

But the way the expo library is presented, you are unable to know if a user clicks submit, so you can only store if the `requestReview` method was called or not.

Shouldn't the library also expose an `onSubmit` callback? Or a boolean that shows if the review was submitted successfully?

What are Expo's thoughts on this?

**Comment by vanGalilea** [comment]
@EvanBacon / @byCedric would you be able to help us with this?

**Comment by Djmill** [comment]
Imo the documentation simply needs to be updated so that developers aren't confused about how this library works. Ex. https://github.com/expo/expo/issues/20553 and https://stackoverflow.com/questions/75656237/expo-store-review-not-working-when-storereview-requestreview were very helpful to understand what the library does under the covers. Why it's not included in the documentation though is beyond me.

**Comment by sohammondal** [comment]
I agree! A `onSubmit` or someway to at least know if the user has previously submitted a review or not.
